### PR TITLE
Consistency for encode/decode behavior.

### PR DESCRIPTION
**Pull request by bbsbb**
Hey there,

Directly PR-in as it's a minor thing. I'm not a fan of encode/decode having different behavior - One defaulting to HS256 and the other throwing an exception. This is a minor change that makes both default to the same algorithm and I const-ed the available algorithms as QoL for foreign code.
Apologies for not discussing first, let me know if it's an issue.

### CODE REVIEW

**Comment by bbsbb** [comment]
Agreement signed.

File: Authentication/JWT.php [hunk 1]
```diff
@@ -35,7 +41,7 @@ class JWT
      *
      * @param string      $jwt           The JWT
      * @param string|Array|null $key     The secret key, or map of keys
-     * @param Array       $allowed_algs  List of supported verification algorithms
+     * @param Array       $allowed_algs  List of supported verification algorithms. Defaults to HS256.
```
Comment (robertdimarco) [inline Authentication/JWT.php]: I'm not inclined to do this. I similarly dislike the asymmetry between the `encode()` and `decode()` methods, but the reason that this is explicit here is to force consumers of this library to be knowledgable about the token algorithms supported in their applications.

The previous weakness here is that a consumer of this library expecting asymmetrically-signed tokens (i.e. verifying the token with a public key, expecting that token to be signed with `RS*`), unknowingly approved tokens signed *symmetrically* (i.e. `HS*) with the same public key. Defaulting to one algorithm or the other seems to improve this issue by making it clearer, but forcing the inclusion of the algorithm in the method signatures aims to make its importance clear to newcomers / maintainers of code using this library.

**Comment by bbsbb** [comment]
Alright, I don't think the rest is worth a PR, so feel free to close. Thanks.

**Comment by robertdimarco** [comment]
@bbsbb Thank you for the time that you put into this PR, and apologies for the trouble there. I hope my rationale was clear, and don't hesitate to reach out if you have other questions or concerns.

**Pull request closed without merging**